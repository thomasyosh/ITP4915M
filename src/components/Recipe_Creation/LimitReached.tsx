import React from 'react';
import { Button } from '@headlessui/react';
import { useRouter } from 'next/router';
import { ExclamationCircleIcon } from '@heroicons/react/24/solid'

interface LimitReachedProps {
    message?: string;
    onAction?: () => void;
    actionText?: string;
    fullHeight?: boolean;
}

const LimitReached: React.FC<LimitReachedProps> = ({
    message = "You've reached your recipe creation limit.",
    onAction,
    actionText = "Go to Home",
    fullHeight = false,
}) => {
    const router = useRouter();

    const handleAction = () => {
        if (onAction) {
            onAction();
        } else {
            router.push('/');
        }
    };

    return (
        <div className={`flex flex-col items-center justify-top ${fullHeight ? 'min-h-screen' : 'h-full'} bg-gray-100 p-4`}>
            <div className="bg-white rounded-lg shadow-lg p-8 max-w-md text-center">
                {/* Icon */}
                <ExclamationCircleIcon className="block m-auto h-16 w-16 text-red-500"/>
                {/* Title */}
                <h2 className="text-2xl font-bold mb-2 text-gray-800">Limit Reached</h2>
                {/* Message */}
                <p className="text-gray-600 mb-6">{message}</p>
                {/* Action Button */}
                <Button
                    onClick={handleAction}
                    className="bg-blue-600 text-white px-4 py-2 rounded-md hover:bg-blue-500 focus:outline-none focus:ring-2 focus:ring-blue-400"
                >
                    {actionText}
                </Button>
            </div>
        </div>
    );
};

export default LimitReached;
