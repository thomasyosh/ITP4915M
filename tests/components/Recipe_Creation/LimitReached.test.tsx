import LimitReached from "../../../src/components/Recipe_Creation/LimitReached";
import { fireEvent, render, screen } from '@testing-library/react'

const routePushMock = jest.fn()

jest.mock("next/router", () => ({
    useRouter: jest.fn(() => ({
        pathName: 'mocked Path',
        push: routePushMock
    })),
}))

describe('The Limit Reached Component', () => {
    it('shall render', () => {
        const { container } = render(<LimitReached />)
        expect(container).toMatchSnapshot()
    })
    it('shall go back home', async () => {
        render(<LimitReached />)
        const homeButton = await screen.findByText('Go to Home')
        fireEvent.click(homeButton)
        expect(routePushMock).toHaveBeenCalledWith('/')
    })
})