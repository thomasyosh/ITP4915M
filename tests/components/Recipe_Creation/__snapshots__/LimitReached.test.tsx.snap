// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Limit Reached Component shall render 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-top h-full bg-gray-100 p-4"
  >
    <div
      class="bg-white rounded-lg shadow-lg p-8 max-w-md text-center"
    >
      <svg
        aria-hidden="true"
        class="block m-auto h-16 w-16 text-red-500"
        data-slot="icon"
        fill="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12ZM12 8.25a.75.75 0 0 1 .75.75v3.75a.75.75 0 0 1-1.5 0V9a.75.75 0 0 1 .75-.75Zm0 8.25a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Z"
          fill-rule="evenodd"
        />
      </svg>
      <h2
        class="text-2xl font-bold mb-2 text-gray-800"
      >
        Limit Reached
      </h2>
      <p
        class="text-gray-600 mb-6"
      >
        You've reached your recipe creation limit.
      </p>
      <button
        class="bg-blue-600 text-white px-4 py-2 rounded-md hover:bg-blue-500 focus:outline-none focus:ring-2 focus:ring-blue-400"
        data-headlessui-state=""
        type="button"
      >
        Go to Home
      </button>
    </div>
  </div>
</div>
`;
